ra/registration-authority.go

```diff
@@ -90,6 +98,45 @@ type certificateRequestEvent struct {
 	Error               string    `json:",omitempty"`
 }
 
+var issuanceCountCacheLife = 30 * time.Minute
+
+// issuanceCountInvalid checks if the current issuance count is invalid either
+// because it hasn't been set yet or because it has expired. This method expects
+// that the caller holds either a R or W ra.tiMu lock.
+func (ra *RegistrationAuthorityImpl) issuanceCountInvalid(now time.Time) bool {
+	return ra.totalIssuedCache == nil || ra.lastIssuedCount.Add(issuanceCountCacheLife).Before(now)
+}
+
+func (ra *RegistrationAuthorityImpl) getIssuanceCount() (*int64, error) {
```

Comment:
deref the count before returning

Avoids having folks munging it by playing with the pointer.